virttest/libvirt_xml/vm_xml.py

```diff
@@ -923,13 +923,45 @@ def get_cpu_topology(self):
             logging.debug("<cpu>/<topology> xml element not found")
         return topology
 
-    def get_disk_all(self):
+    def get_disk_all(self, disk_type=None, disk_device=None, reverse=False):
         """
         Return VM's disk from XML definition, None if not set
+
+        There is a issue that when the disks different bus type, they may have
+        same target dev value, so the previous disk will be overwirten by the
+        last one. In order to avoid this problem, you could specify the device
+        value by 'disk_device' to only select the desired disks or the disks
+        you don't want by 'reverse' and 'disk_device'.
+
+        This xml will not return all disks by default paramenters:
+        <disk type='file' device='disk'>
+          <source file='[esx6.7-matrix] xxx/xxx-xxx.vmdk'/>
+          <target dev='sda' bus='scsi'/>
+          <address type='drive' controller='0' bus='0' target='0' unit='0'/>
+        </disk>
+        <disk type='file' device='cdrom'>
+          <target dev='sda' bus='sata'/>
+          <address type='drive' controller='0' bus='0' target='0' unit='0'/>
+        </disk>
+
+        :param disk_type: type attribute value of the disk
+        :param disk_device: device attribute value of the disk
+        :param reverse: If true, return disks that are not equal the specified
+            type or device attribute.
         """
         disk_nodes = self.xmltreefile.find('devices').findall('disk')
         disks = {}
         for node in disk_nodes:
+            if reverse:
+                if disk_type and node.get('type') == disk_type:
+                    continue
+                if disk_device and node.get('device') == disk_device:
+                    continue
+            else:
+                if disk_type and node.get('type') != disk_type:
+                    continue
+                if disk_device and node.get('device') != disk_device:
+                    continue
```

Comment:
Actually @xiaodwan , I'm wondering, could we create another function of getting disks and keep this old one at the same time?  Then we could have a better design of the function without worrying affecting other tests.

Reply:
And when other test cases met the same problem, they could switch to the new function if that's what they want.

Reply:
The change is compatible with existing codes. But anyway, if all agree to create a new function, I will do that.
Thanks.

Reply:
:handshake:

Reply:
I agree with writing a new function.